### ISSUE TITLE
Add -CfVFPV3 when compiling for Android

### ISSUE THREAD

**Issue by michaliskambi** [newpascal/fpcupdeluxe#31]
Initially when handling https://github.com/newpascal/fpcupdeluxe/pull/10 , you have set these options for modern Android devices:

~~~~
FPCupManager.CrossOPT:='-CpARMV7A -CfVFPV3 '
~~~~

However, in the commit https://github.com/newpascal/fpcupdeluxe/commit/b3351e169d90574271d8f6bd5d93bb07c9823e6c , you changed them to

~~~~
FPCupManager.CrossOPT:='-CpARMV7A '
~~~~

The change is in `fpcupdeluxemainform.pas`:

~~~~
         if (FPCupManager.CrossOS_Target='android')
-            then FPCupManager.CrossOPT:='-CpARMV7A -CfVFPV3 '
+            then FPCupManager.CrossOPT:='-CpARMV7A '
~~~~

This disables using hardware floats, which makes all floating-point calculations very sluggish (this results in around 4x performance drop when using Castle Game Engine, see https://github.com/castle-engine/castle-engine/wiki/Android-FAQ#notes-about-compiling-with-hard-floats--cfvfpv3 ). As far as I know, all modern devices in practice support `-CfVFPV3`.

The `settings.ini` (not used by fpcupdeluxe) also have `crossopt="-CpARMV7 -CfVFPV3" ` in `[android]` section.

Can you add -CfVFPV3 back? If there's a reason why you don't want to use it by default, then all CGE users will have to adjust these options (thankfully, it's easy now thanks to #13  :)

**Comment by LongDirtyAnimAlf** [comment]
I will put it back. If there are no alternatives.

The reason to remove:
I got a request for an arm toolchain for Mac (Darwin).
This armv8 toolchain was not happy with -CfVFPV3.
But diving into the docs it seems that armv8 always has VFPv3/v4 enabled ?
Would you mind to comment ?

**Comment by michaliskambi** [comment]
I don't really know about arm8, or Darwin, that much...

The one thing I know is that for Android, the arm7 is used, and the -CfVFPV3 is *not* available by default. And miixing code compiled with and without -CfVFPV3 is not possible.

**Comment by LongDirtyAnimAlf** [comment]
Thanks. Will revert in a few hours !
And close if done.

Request: if this is done, would you mind updating your wiki ?
If done, I will put a link towards your wiki from the fpcupdeluxe wiki page !

**Comment by michaliskambi** [comment]
> Request: if this is done, would you mind updating your wiki ?
> If done, I will put a link towards your wiki from the fpcupdeluxe wiki page !

Absolutely! I will happily update https://github.com/castle-engine/castle-engine/wiki/fpcupdeluxe then:) I *want* everyone to use the original version of fpcupdeluxe.

**Comment by LongDirtyAnimAlf** [comment]
Please test.
https://github.com/newpascal/fpcupdeluxe/releases/tag/1.4.0

**Comment by michaliskambi** [comment]
> Please test.

Tested, everything is perfect:) Thank you!

I also tested getting CGE from GitHub using fpcupdeluxe, works perfectly too.

I see that you made a lot of other improvements to fpcupdeluxe lately, this is very very much appreciated:) Thank you!

I updated https://github.com/castle-engine/castle-engine/wiki/fpcupdeluxe . Feel free to edit that wiki page too, if you find there something outdated or incorrect (it should be publicly editable -- let me know if it's not).
